tfexec/add.go

```diff
@@ -0,0 +1,88 @@
+package tfexec
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+	"strconv"
+)
+
+type addConfig struct {
+	fromState       bool
+	out             string
+	includeOptional bool
+	provider        string
+	reattachInfo    ReattachInfo
+}
+
+var defaultAddOptions = addConfig{}
+
+type AddOption interface {
+	configureAdd(*addConfig)
+}
+
+func (opt *FromStateOption) configureAdd(conf *addConfig) {
+	conf.fromState = opt.fromState
+}
+
+func (opt *OutOption) configureAdd(conf *addConfig) {
+	conf.out = opt.path
+}
+
+func (opt *IncludeOptionalOption) configureAdd(conf *addConfig) {
+	conf.includeOptional = opt.includeOptional
+}
+
+func (opt *ProviderOption) configureAdd(conf *addConfig) {
+	conf.provider = opt.provider
+}
+
+func (opt *ReattachOption) configureAdd(conf *addConfig) {
+	conf.reattachInfo = opt.info
+}
+
+// Add represents the terraform add subcommand.
+func (tf *Terraform) Add(ctx context.Context, address string, opts ...AddOption) error {
```

Comment:
```suggestion
func (tf *Terraform) Add(ctx context.Context, address string, opts ...AddOption) (string, error) {
```

The `add` command writes configuration to stdout by default and so I think that the interface of this method should reflect that default behaviour and probably just entirely omit support for the `out` flag.

IMHO in contexts where `terraform-exec` is used users/developers can still process the string and write it to a file if they choose to.

Reply:
Indeed, this is totally a mistake. Thank you for pointing this out :+1: